### ISSUE TITLE
FEATURE: can always_show by pattern

### ISSUE THREAD

**Issue by jcarstairs-scottlogic** [nvim-neo-tree/neo-tree.nvim#1255]
### Did you check the docs?

- [X] I have read all the docs.

### Is your feature request related to a problem? Please describe.

I want to be able to always show environment files, since I often want to edit these, but they're also commonly listed in a `.gitignore`, so they are often hidden by Neotree. Environment files generally follow a simple pattern: `.env*`. But right now, I can only list possible literal environment file names one by one.

### Describe the solution you'd like.

The `always_show` property should accept glob patterns, the same as the `hide_by` property.

To me, it looks like this means there should be a function call something like `utils.is_filtered_by_pattern(f.always_show, path, name)` [when file items get created](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/9298c1b7a52e63ed4fd977fd792a0d4afe9260d5/lua/neo-tree/sources/common/file-items.lua#L167C8-L167C9). Compare [how the hide_by property is handled](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/9298c1b7a52e63ed4fd977fd792a0d4afe9260d5/lua/neo-tree/sources/common/file-items.lua#L175).

### Describe alternatives you've considered.

I could just list environment file names. But that's going to get annoying quickly. Realistically, I'll never bother.

**Comment by cseickel** [comment]
That seems perfectly reasonable. If you or anyone else wants to put up a PR for an `always_show_by_pattern` config I will accept it.

**Comment by PumpedSardines** [comment]
I made a PR for this feature request since i needed it myself

**Comment by dpetka2001** [comment]
Does the PR that was made actually work? I tried `always_show_by_pattern = { ".git*" }` and it doesn't always show the directories `.git`, `.github` and the file `.gitignore` by default. Setting `always_show = { ".git", ".github", ".gitignore" }` does work as intended. Am I doing something wrong when using `always_show_by_pattern`?

**Comment by PumpedSardines** [comment]
@dpetka2001 just tried adding `always_show_by_pattern = { ".git*" }` to my config and it works on my setup. I see all folders and files starting in `.git`. Are you sure you're using the right version of neo-tree?

Screenshot showing my default view with `always_show_by_pattern = { ".git*" }`
<img width="215" alt="image" src="https://github.com/nvim-neo-tree/neo-tree.nvim/assets/58148984/34ca9fb2-3e64-44f0-91f5-ce489c95451a">

**Comment by dpetka2001** [comment]
Yes, I had `branch = "v3.x"`. Removed that and was able to update to latest HEAD. Thanks for the heads up. Now it works as it should :smile:
